xray/src/lib.rs

```diff
@@ -44,6 +47,18 @@ impl Meta {
         Ok(Self::from_proto(&proto))
     }
 
+    pub fn to_disk<P: AsRef<Path>>(&self, filename: P) -> io::Result<()> {
```

Comment:
Can we also use this method in `generation.rs`?

Reply:
done